rest/config.go

```diff
@@ -331,9 +331,76 @@ func (dbConfig *DbConfig) setup(name string) error {
 		dbConfig.Server = &urlStr
 	}
 
+	// If the sync function is referenced from an external file or http(s) endpoint
+	// unlike the inline JavaScript definition, read it from the specified path.
+	if dbConfig.Sync != nil {
+		sync, err := loadJavaScript(*dbConfig.Sync)
+		if err != nil {
+			return err
+		}
+		dbConfig.Sync = base.StringPtr(sync)
+	}
+
+	// If the import filter function is referenced from an external file or http(s) endpoint
+	// unlike the inline JavaScript definition, read it from the specified path.
+	if dbConfig.ImportFilter != nil {
+		importFilter, err := loadJavaScript(*dbConfig.ImportFilter)
+		if err != nil {
+			return err
+		}
+		dbConfig.ImportFilter = base.StringPtr(importFilter)
+	}
+
+	// If the conflict resolution function is referenced from an external file or http(s) endpoint
+	// unlike the inline JavaScript definition, read it from the specified path.
+	for _, rc := range dbConfig.Replications {
+		if rc.ConflictResolutionFn != "" {
+			conflictResolutionFn, err := loadJavaScript(rc.ConflictResolutionFn)
+			if err != nil {
+				return err
+			}
+			rc.ConflictResolutionFn = conflictResolutionFn
+		}
+	}
+
 	return err
```

Comment:
This `err` is from `url.Parse` way up on line 318.

Should we return nil at the end of this function, and an extra `if err != nil { return err }` above the loadJavaScript code for `url.Parse`?